src/dtgtk/culling.c

```diff
@@ -1044,9 +1047,6 @@ void dt_culling_init(dt_culling_t *table, int fallback_offset)
       }
       sqlite3_finalize(stmt);
     }
-
-    // we now determine if we limit culling images to the selection
-    table->navigate_inside_selection = (!table->selection_sync && inside);
```

Comment:
fyi, it's here that we currently "automaticaly" decide the restrict-unrestrict mode...
otherwise the code seems quite clean and not too intrusive...